spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryProperties.java

```diff
@@ -159,19 +164,36 @@
 
 	/**
 	 * Map of serviceId's -> tag to query for in server list. This allows filtering
-	 * services by a single tag.
+	 * services by a single tag. (Deprecated - instead use 'serviceInstanceQueryTags')
 	 */
+	@Deprecated
 	private Map<String, String> serverListQueryTags = new HashMap<>();
 
+	/**
+	 * Tag to query for in service list if one is not listed in serverListQueryTags.
+	 * (Deprecated - instead use 'defaultServiceInstanceQueryTags')
+	 */
+	@Deprecated
+	private String defaultQueryTag;
+
+	/**
+	 * Map of serviceId -> tags to use when querying for service instances. This allows
+	 * filtering service instances by tags.
+	 */
+	private Map<String, List<String>> serviceInstanceQueryTags = new HashMap<>();
+
+	/**
+	 * Tags to use when querying for service instances if the server id in question does
+	 * not have an entry in 'serviceInstanceQueryTags'
+	 */
```

Comment:
Use `@link` instedad of '' in javadocs.

Reply:
I usually do that but have learned to **not** do that when in Spring Boot as the post processor that generates the docs from the `@ConfigurationProperties` does not handle that well (IIRC). Is that not the case here?